samza-core/src/main/scala/org/apache/samza/container/TaskInstanceMetrics.scala

```diff
@@ -37,7 +37,14 @@ class TaskInstanceMetrics(
   val flushes = newCounter("flush-calls")
   val pendingMessages = newGauge("pending-messages", 0)
   val messagesInFlight = newGauge("messages-in-flight", 0)
-  val asyncCallbackCompleted = newCounter("async-callback-complete-calls");
+  val asyncCallbackCompleted = newCounter("async-callback-complete-calls")
+
+  val asyncCommitSkipped = newGauge("async-commits-skipped", 0)
+  val asyncUploadsCompleted = newCounter("async-uploads-completed")
+  val asyncUploadNs = newTimer("async-upload-ns")
+  val commitNs = newTimer("commit-ns")
```

Comment:
Minor: order these hierarchically: commitNs -> [snapshotNs, asyncCommitNs -> [uploadNs, checkpointWriteNs, cleanupNs]]

Reply:
Does it make sense to have a container wide version of these metrics for an overview as well (in addition to per task version)?